tests/integration/states/test_file.py

```diff
@@ -888,6 +888,17 @@ def test_directory(self):
         self.assertSaltTrueReturn(ret)
         self.assertTrue(os.path.isdir(name))
 
+    def test_directory_issue_50610(self):
+        '''
+        file.directory, keep casing
+        '''
+        dirname = 'a_NeW_DiR_KeEp_CaSiNg'
+        name = os.path.join(RUNTIME_VARS.TMP, dirname)
+        ret = self.run_state('file.directory', name=name)
+        self.assertSaltTrueReturn(ret)
+        self.assertTrue(dirname in os.listdir(RUNTIME_VARS.TMP))
+        self.assertTrue(os.path.isdir(name))
+
```

Comment:
Could you add this test to the 2017.7 branch? It will be merged forward into later branches. I believe this issue has been fixed by https://github.com/saltstack/salt/pull/51757